subsys/bluetooth/mesh/cfg.c

```diff
@@ -79,6 +82,15 @@ static enum bt_mesh_feat_state feature_get(int feature_flag)
 		       BT_MESH_FEATURE_DISABLED;
 }
 
+static int check_proxy_adv(struct bt_mesh_subnet *sub, void *cb_data)
+{
+	if (sub->node_id == BT_MESH_NODE_IDENTITY_RUNNING) {
+		return 1;
+	}
+
+	return 0;
+}
```

Comment:
> Shouldn't it be `sub->node_id == BT_MESH_NODE_IDENTITY_STOPPED;` since it is stopped when network id is advertised? Otherwise if `node_id` is `BT_MESH_NODE_IDENTITY_RUNNING` for at least one subnet, the advertiser won't be updated.

I believe this is the point - we only want to trigger this code if the advertiser is currently waiting on a new mesh packet?

Reply:
Then it should check whether the currently advertising subnet advertises network id or not.

Reply:
I'm pretty sure it should check `BT_MESH_NODE_IDENTITY_RUNNING` state. When an advertiser is in this state it means Node ID is in progress. This advertising should be completed. 
```
7.2.2.2.1 Advertising
A node may advertise with Node Identity regardless of the Proxy feature being
supported or enabled.
``` 
That means the device shouldn't touch ongoing advertising. The only state changing will be sufficient. The afterword switching to adv thread will consider state and stop Network ID advertising. 

And the opposite situation if no one subnetwork is advertising Node ID - all are 'BT_MESH_NODE_IDENTITY_STOPPED'. That means Network ID is in progress. `bt_mesh_adv_update` will skip the FIFO waiting time and initiate the next advertising session. That causes immediate stopping of Network ID advertising.

Reply:
Yep, but this information is not available here. There is only `static struct bt_mesh_subnet *beacon_sub;` in `proxy_srv.c` that contains pointer after the used here pointer. I don't think this is correct to provide access to the encapsulated pointer. That creates hidden interdependency between components. IMHO the usage of internal API is the only correct way.

Reply:
Couldn't it be solved with a new function in `proxy_srv.c`?

Reply:
Really do not see advantages in creating the single case API for the very specific situation. Not a big load to walk through subnetworks to check any of them considering that the disabling\enabling functionality like GATT proxy is a quite rare case.